internal/comparetest/metrics.go

```diff
@@ -188,21 +196,25 @@ func CompareMetricSlices(expected, actual pmetric.MetricSlice) error {
 // CompareNumberDataPointSlices compares each part of two given NumberDataPointSlices and returns
 // an error if they don't match. The error describes what didn't match.
 func CompareNumberDataPointSlices(expected, actual pmetric.NumberDataPointSlice) error {
-	if expected.Len() != actual.Len() {
-		return fmt.Errorf("number of datapoints does not match expected: %d, actual: %d", expected.Len(), actual.Len())
+	exp, act := pmetric.NewNumberDataPointSlice(), pmetric.NewNumberDataPointSlice()
+	expected.CopyTo(exp)
+	actual.CopyTo(act)
+
```

Comment:
Should we keep the copied slices here, even though there is no sorting function?

Reply:
No, it's not needed here